flexget/components/trakt/trakt_calendar.py

```diff
@@ -86,54 +89,65 @@ class TraktCalendar:
     @cached('trakt_calendar', persist='2 hours')
     def on_task_input(self, task, config):
         start_date = datetime.datetime.now().date() + datetime.timedelta(days=config['start_day'])
-        logger.debug(
-            'Start date for calendar: {}, end date: {}',
-            start_date,
-            start_date + datetime.timedelta(days=config['days']),
-        )
-
-        url = db.get_api_url(
-            'calendars',
-            'my' if config.get('account') else 'all',
-            'shows',
-            start_date,
-            config['days'],
-        )
-
-        try:
-            results = (
-                db.get_session(config.get('account')).get(url, params={'extended': 'full'}).json()
-            )
-            logger.debug('Found {} calendar entries', len(results))
-        except RequestException as e:
-            raise plugin.PluginError('Error while fetching calendar: {0}'.format(e))
 
+        # The API limit is max_number_of_days days for a single call. Find the number of calls.
+        number_of_calls = math.ceil(config['days'] / max_number_of_days)
+        days = config['days']
         entries = set()
-        for result in results:
-            e = Entry()
-            e.update_using_map(self.series_map, result['show'])
-            if config['type'] == 'episodes':
-                e.update_using_map(self.episode_map, result['episode'])
 
-            title = e['trakt_series_name']
-            if not config['strip_dates']:
-                title = '{0} ({1})'.format(title, e['trakt_series_year'])
+        for _ in range(number_of_calls):
```

Comment:
Probably fine the way you have it, something like this way feels a bit more pythonic to me though.
```
for start_day in range(0, config['days'], 31):
    num_days = min(config['days'] - start_day, 31)
```